cmd/ndm_daemonset/probe/capacityprobe.go

```diff
@@ -92,16 +95,15 @@ func (cp *capacityProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDe
 		klog.Error("unable to parse the block size", err)
 		return
 	}
-	b, err = ioutil.ReadFile(sysPath + "/queue/hw_sector_size")
-	if err != nil {
-		klog.Error("unable to read sector size", err)
-		return
-	}
-	sectorSize, err := strconv.ParseInt(strings.TrimSuffix(string(b), "\n"), 10, 64)
-	if err != nil {
-		klog.Error("unable to parse the sector size", err)
-		return
-	}
+
+	// The size (/size) entry returns the `nr_sects` field of the block device structure.
+	// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L1267
+	//
+	// Traditionally, in Unix disk size contexts, “sector” or “block” means 512 bytes,
+	// regardless of what the manufacturer of the underlying hardware might call a “sector” or “block”
+	// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L487
+	//
+	// Therefore, to get the capacity of the device it needs to always multiplied with 512
```

Comment:
nice, so this was a bug. On google cloud with local attached SSD, we will see higher size that the disk size attached.

Reply:
yes. But we haven't faced the issue because the value that was filled in was over written by smartprobe(because capacity-probe had lowest priority). If smart probe is disabled, the value fetched by capacity probe for local SSDs were wrong.

Reply:
There must be new devices that show the sectorsize as 4K. ( cc: @richardelling ) 

Also, just need to make sure we are not re-opening the issue fixed here: https://github.com/openebs/node-disk-manager/pull/150

Reply:
New devices(SSDs) show `hw_sector_size` as 4k. But to get the capacity of the device, we need to always multiply the value in `/sys/class/block/sda/size` with 512. 

This is because the `size` file contains the no.of blocks on the block device and it is always calculated by using 512 byte  irrespective of the hardware sector size of the device